core/consensus/babe/impl/babe_block_validator_impl.cpp

```diff
@@ -94,6 +127,10 @@ namespace kagome::consensus::babe {
              epoch_number,
              config.randomness);
 
+    if (babe_header.authority_index >= config.authorities.size()) {
+      return ValidationError::NO_VALIDATOR;
```

Comment:
Maybe worth leaving a warning log explaining that validator's index is out of range